src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -218,35 +218,13 @@ RadioSpinel<InterfaceType, ProcessContextType>::RadioSpinel(void)
 }
 
 template <typename InterfaceType, typename ProcessContextType>
-void RadioSpinel<InterfaceType, ProcessContextType>::Init(bool aResetRadio,
-                                                          bool aRestoreDatasetFromNcp,
-                                                          bool aSkipRcpCompatibilityCheck)
+void RadioSpinel<InterfaceType, ProcessContextType>::Init(bool aRestoreDatasetFromNcp, bool aSkipRcpCompatibilityCheck)
```

Comment:
The `no-reset` option is used to suppress sending reset command when the SPI interface is used (See https://github.com/openthread/openthread/pull/3800). Because the SPI interface will trigger the hardware reset, it doesn't need to do the software reset. Removing the `no-reset` won't affect the compatibility between and the host and RCP.

The name of the `no-reset` is not right, it should be `no-soft-reset`. I think it is better to remove it.

If developers have already configured the option `no-reset`, after we remove the `no-reset` option, the Thread stack still could work without changing the configuration. From this point of view, it won't affect others. 

Thoughts @abtink @bukepo

Reply:
I think with this change in this PR we now always reset twice in case of SPI interface? SPI resets, and then we sent a soft reset?. 

I understand it may not impact compatibility, but is it a good idea to do this?

Reply:
The name `no-reset` was added when there's only software reset in OpenThread POSIX before spi-hdlc-adapter's functionality was integrated in. `no-reset` is used for SPI RCP and simulation RCP. I'm inclined to keep it.

Reply:
@abtink I have updated the code to add an API `Reset(aResetType)`. Software reset or hardware reset chip will be determined by the implementation of the interface. Could you help review it again?

@bukepo I have updated the code to keep the `no-reset`  option.